charmhelpers/fetch/ubuntu.py

```diff
@@ -457,7 +545,55 @@ def _add_apt_repository(spec):
     if '{series}' in spec:
```

Comment:
As this function is being modded, it's a good opportunity to add the `:type spec:` to the docstring, please.